util/AbilityAbstraction.lua

```diff
@@ -214,11 +214,12 @@ M.Concat = function(self, a, ...)
     rec(a, ...)
     return g
 end
+-- Remove items matching b from array a
```

Comment:
The modified behavior is better and no longer depends on the caller's handling of nil, thanks.

Reply:
By the way, I find the method of deleting table elements in Lua really weird. Take a look when you have time: https://stackoverflow.com/questions/1758991/how-to-remove-a-lua-table-entry-by-its-key

Reply:
I think setting a key's associated value to nil makes a lot of sense. But the fact that an "array" is just the elements of a table with sequential integer keys starting at 1 definitely cooks the brain a bit.

If we had wanted to use a copy-and-remove approach, then we could have used table.remove(i), which would have properly shifted the other members of the array instead of creating a nil gap.  That approach would have left the non-array members unchanged (even if they contained the value being removed), while the current approach keeps only the array members.